tests/Microsoft.DotNet.Docker.Tests/StaticTagTests.cs

```diff
@@ -310,12 +259,288 @@ public void VersionTag_SameOsAndVersion(Repo repo, int versionParts)
+                            });
+                        }
+                        testObjects.AddRange(new[]
+                        {
+                            GetTagTestInput(
+                                testType, 
+                                repo, 
+                                versionType: VersionType.MajorMinor, 
+                                checkArchitecture: true
+                            ), // <Major.Minor>-alpine-<architecture>
+
+                            GetTagTestInput(
+                                testType, 
+                                repo, 
+                                versionType: VersionType.MajorMinor, 
+                                checkArchitecture: false
+                            ) // <Major.Minor>-alpine
+                        });
+                        break;
+
+                    case TestType.Version:
+                        if (ApplianceRepos.Any(repo.Name.Contains))
+                        {
+                            // Only appliance repos have major version tags
+                            testObjects.Add(GetTagTestInput(
+                                testType,
+                                repo,
+                                versionType: VersionType.Major
+                            )); // <Major>
+                        }
+                        testObjects.AddRange(new[]
+                        {
+                            GetTagTestInput(
+                                testType,
+                                repo,
+                                versionType: VersionType.MajorMinor
+                            ), // <Major.Minor>
+
+                            GetTagTestInput(
+                                testType,
+                                repo,
+                                versionType: VersionType.MajorMinorPatch
+                            ), // <Major.Minor.Patch>
+                        });
+                        break;
+
+                    default:
+                        throw new ArgumentException("Invalid tag type", nameof(testType));
+                }
+            }
+            return testObjects.ToArray();
+        }
+
+        private static object[] GetTagTestInput(
+            TestType testType,
+            Repo repo,
+            VersionType? versionType = null,
+            bool? checkOs = false,
+            bool? checkArchitecture = false,
+            Func<ManifestHelper.DockerfileInfo, bool>? skipDockerfileOn = null)
+        {
+            switch (testType)
+            {
+                case TestType.Latest:
+                    return [ repo ];
+
+                case TestType.Platform:
+                    if (versionType == null || checkOs == null || checkArchitecture == null)
+                    {
+                        throw new ArgumentException("'versionType', 'checkOs' and 'checkArchitecture' must be specified", nameof(testType));
```

Comment:
Follow this pattern for the other exceptions below too.

```suggestion
                        throw new ArgumentException($"'{nameof(versionType)}', '{nameof(checkOs)}' and '{nameof(checkArchitecture)}' must be specified", nameof(testType));
```